lib/floe/runner/container_runner.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+require_relative "container_runner/docker_mixin"
+require_relative "container_runner/docker"
+require_relative "container_runner/kubernetes"
+require_relative "container_runner/podman"
+
+module Floe
+  class Runner
+    class ContainerRunner
+      class << self
+        def cli_options(optimist)
+          optimist.banner("")
+          optimist.banner("Container runner options:")
+
+          optimist.opt :container_runner, "Type of runner for docker container images (docker, podman, or kubernetes)", :type => :string, :short => 'r'
+          optimist.opt :container_runner_options, "Options to pass to the container runner", :type => :strings, :short => 'o'
+
+          optimist.opt :docker,     "Use docker to run container images     (short for --container_runner=docker)",     :type => :boolean
+          optimist.opt :podman,     "Use podman to run container images     (short for --container_runner=podman)",     :type => :boolean
+          optimist.opt :kubernetes, "Use kubernetes to run container images (short for --container_runner=kubernetes)", :type => :boolean
+        end
+
+        def resolve_cli_options(opts)
```

Comment:
Looks like this modifies the opts being passed in, how about either `resolve_cli_options!` or maybe call this `set_container_runner`

Reply:
oh yeah good idea! on the `!`

I wanted a generically named interface that could be called from exe/floe.  So, however the plugin wants to deal with the cli options is up to them.